TESTS/network/wifi/main.cpp

```diff
@@ -74,6 +74,7 @@ Case cases[] = {
     Case("WIFI-GET-RSSI", wifi_get_rssi),
     Case("WIFI-CONNECT-PARAMS-VALID-UNSECURE", wifi_connect_params_valid_unsecure),
     Case("WIFI-CONNECT", wifi_connect),
+    //Case("WIFI-CONNECT-NONBLOCKING", wifi_connect_nonblock),
```

Comment:
this should not be in here (can be added anytime).

Why is it commented out ? The implementation is being added in this PR.

Reply:
In fact, none of the Wi-Fi class implementations we checked is able to pass this test right now. We want the test to be available, however, in order to determine the expected behaviour and be able to ask Wi-Fi stack developers (Ublox, ST, etc.) to implement the missing functionality or at least return `NSAPI_ERROR_UNSUPPORTED`, so the test can be skipped.
We thought it would be easiest for someone starting the implementation with a test-driven-development to uncomment this line, run the test and try to make it pass. 
Removing the test is also fine - most developers will probably be able to figure out how to run it ;-). 
I would say, however, that leaving a commented test will not let us (and anyone developing a W-Fi driver) forget about this function.
@0xc0170 , how about we add a comment explaining the situation, like:
```
// Most boards are not passing this test, but they should if they support non-blocking API.
```
?

Reply:
The comment would be better than it is now.